### ISSUE TITLE
VRAM Allocations on Ampere

### ISSUE THREAD

**Issue by XDynames** [facebookresearch/detectron2#2133]
There seems to be a disparity between the max memory consumption reported by pytorch and the observed memory used in nvidia-smi. This may be a problem with Ampere specifically as prior to install the 3090 I felt as though these numbers where quite consistent. It also seems to climb - like there is something not being deallocated (The climbing behaviour persists for longer than the normal ramp up observed as the RPN settles in)

![Screen Shot 2020-10-15 at 9 47 57 am](https://user-images.githubusercontent.com/39457784/96055418-8821d280-0ecb-11eb-802c-1feb05441a35.png)

## Instructions To Reproduce the Issue:
Install pytorch using cuda 11 binary found [here](https://discuss.pytorch.org/t/rtx-3000-support/98158l)
Build detectron2 from source
Run mask RCNN train

## Expected behavior:
Max memory displayed by tqdm to be reflected in the VRAM allocation show to python in nvidia-smi

## Environment:

Provide your environment information using the following command:
```
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.0
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80
  - CuDNN 8.0.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 
```

**Comment by ppwwyyxx** [comment]
>  disparity between the max memory consumption reported by pytorch and the observed memory used in nvidia-smi

A disparity is expected due to how the memory allocator works: https://pytorch.org/docs/stable/notes/cuda.html#memory-management

However, if the disparity is too large, it might indicate pytorch's allocator is inefficient and you can report it to pytorch. Including information like `torch.cuda.memory_stats` or `torch.cuda.memory_summary()` there might help understand what is wrong.
